crates/sui-indexer/tests/integration_tests.rs

```diff
@@ -55,6 +63,112 @@ mod pg_integration {
         drop(handle);
     }
 
+    #[tokio::test]
+    async fn test_simple_transaction_e2e() -> Result<(), anyhow::Error> {
+        let (test_cluster, indexer_rpc_client, store, _handle) = start_test_cluster().await;
+        // Allow indexer to sync genesis
+        wait_until_next_checkpoint(&store).await;
+        let address = test_cluster.accounts.first().unwrap();
+        let recipient_address = test_cluster.accounts.last().unwrap();
+        let gas_objects: Vec<ObjectID> = indexer_rpc_client
+            .get_owned_objects(
+                *address,
+                Some(SuiObjectDataOptions::new().with_type()),
+                None,
+                None,
+                None,
+            )
+            .await?
+            .data
+            .into_iter()
+            .filter_map(|object_resp| match object_resp {
+                SuiObjectResponse::Exists(obj_data) => Some(obj_data.object_id),
+                _ => None,
+            })
+            .collect();
+
+        let transaction_bytes: TransactionBytes = indexer_rpc_client
+            .transfer_object(
+                *address,
+                *gas_objects.first().unwrap(),
+                Some(*gas_objects.last().unwrap()),
+                1000,
+                *recipient_address,
+            )
+            .await?;
+
+        let keystore_path = test_cluster.swarm.dir().join(SUI_KEYSTORE_FILENAME);
+        let keystore = Keystore::from(FileBasedKeystore::new(&keystore_path)?);
```

Comment:
hmm on vscode i don't get a complaint but in clion w/ rust I get a `mismatched types [E0308]expected Keystore, found FileBasedKeystore`. curious if you're seeing the same?

Reply:
I did not. they are using different language services, VS code prob uses rust-analyzer and not sure about CLion, in my experience, these languages are not super reliable, if it is indeed type errors, you will see it from `cargo build`.